katello/katello/hostname-change.rb

```diff
@@ -343,14 +339,24 @@ def run
       self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' /etc/hosts")
 
       STDOUT.puts "updating hostname in foreman installer scenarios"
-      self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' /etc/foreman-installer/scenarios.d/*.yaml")
+      self.run_cmd("sed -i -e 's/#{@old_hostname}/#{@new_hostname}/g' #{scenarios_path}/*.yaml")
+
+      if File.exist?(last_scenario_yaml)
+        STDOUT.puts 'backing up last_scenario.yaml'
+        temp_last_scenario_yaml = Tempfile.new('last_scenario')
+        begin
+          temp_last_scenario_yaml << File.read(last_scenario_yaml)
+        ensure
+          temp_last_scenario_yaml.close
+        end
 
-      STDOUT.puts "removing last_scenario.yml file"
-      self.run_cmd("rm -rf /etc/foreman-installer/scenarios.d/last_scenario.yaml")
+        STDOUT.puts 'removing last_scenario.yaml'
+        run_cmd("rm -f #{last_scenario_yaml}")
```

Comment:
Isn't this equal to ```File.unlink```?

Reply:
If last_scenario is always a symlink then I'll definitely change this!